lib/elixir/lib/enum.ex

```diff
@@ -1284,7 +1284,7 @@ defmodule Enum do
       "2 = 4 = 6"
 
   """
-  @spec map_join(t, String.t, (element -> any)) :: String.t
+  @spec map_join(t, String.t, (element -> String.t)) :: String.t
```

Comment:
For reference, https://github.com/elixir-lang/elixir/blob/5bd1e0c1e1f3dca7152356b735ad48664d6daab5/lib/elixir/lib/protocol.ex#L509, a protocol type is defined as `t :: term`, so it just serves for documentation purposes, but still worth having :).